integration-tests/velocity/src/main/resources/application.properties

```diff
@@ -15,4 +15,6 @@
 ## limitations under the License.
 ## ---------------------------------------------------------------------------
 
-quarkus.native.resources.includes = template/*
\ No newline at end of file
+quarkus.native.resources.includes = template/*
+
+quarkus.camel.native.reflection.include-patterns=org.apache.camel.Exchange
```

Comment:
Is the reflective config for `Exchange` definitely needed? Just curious how we don't need this with the other templating components.

Reply:
yeah, I agree and is there any issue if this reflective config is not added? `velocityContext` test will be failing?

Reply:
As you mentioned it, I agree that this should work automatically without registration. (At least from the PoV of user this is internal stuff)

Without registration of the Exchange, the execution fails in native mode, because I'm using exchange as an object in template (see `Dear ${headers.name}. You ordered item ${exchange.properties.item} on ${body}`).

```
VelocityTest.testVelocityContext:157 expected: <
Dear Sheldon. You ordered item Earth 1 on Monday.> but was: <
Dear Sheldon. You ordered item ${exchange.properties.item} on Monday.>
```

I was looking for an automatic registration for reflections of internal classes, but I was able to find only classes for serialization - https://github.com/apache/camel-quarkus/blob/main/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelSerializationProcessor.java#L86

I'd say that the similar mechanism would make sense also for registration for reflection and that the `org.apache.camel.Exchange` should be there. What do you think @jamesnetherton , @zhfeng ?
(I can create separate ticket and work on it)

Reply:
Other option is to register `org.apache.camel.Exchange` only in velocity extension (in deployment module)

Reply:
Thanks @JiriOndrusek for explanations and I think it should only be in velocity extension.

Reply:
Probably it's needed because the exchange properties are not handled here:

https://github.com/apache/camel/blob/985ea1061e9d2c1db81c9e6da751258700454978/core/camel-support/src/main/java/org/apache/camel/support/ExchangeHelper.java#L487-L505

In the template, it'd nicer if you could do:

```
Dear ${headers.name}. You ordered item ${properties.item} on ${body}.
```

I.e not have to reference the `Exchange` class. So we should open a Camel ticket and add that capability if possible.

I guess we could live with the workaround in the test until then. I don't really want us to promote reflective access to `Exchange` etc. We already [discourage](https://github.com/apache/camel-quarkus/blob/5127a3c6fa25d4788b2ffddca220545df08cd42e/docs/modules/ROOT/pages/reference/extensions/velocity.adoc#allowcontextmapall-option-in-native-mode) this.

Reply:
That makes sense, thanks for the suggestion. I'll create an improvement ticket on camel to allow ${properties.item}.
Not sure whether the following feature will make sense, but it would be nice, if camel can handle `${exchange.properties.item}` internally as `${properties.item}`. Because user could use `${exchange.properties.item}`

Reply:
I think `${properties.item}` does not work. only `headers` and `body` can be used in templates.

Reply:
If I summarize the whole problem. We suggest to users to not use exchange or context as body  used in templates. Therefore the test should be probably only changed to use different property (to work without the registration of reflection).

Option for ${properties.item) would be a nice improvement on came side.

Reply:
@jamesnetherton would it make sense to register ${properties} only for camel-quarkus? (I think that making it accessible in that way in camel is does not make sense because there is no limitations with exchange, thus there is no need to implement it `${exchange.properties}` is OK from camel's PoV)

Reply:
> would it make sense to register ${properties} only for camel-quarkus

I think we should avoid that. It should work the same across all of the runtimes. If you access headers without referencing the exchange, then to me it seems reasonable to do the same with properties.

Reply:
ok, then I'll change tests to not use this and propose an improvement on camel side -> we will see wheter it will be accepted.

Reply:
is it possible to only register the custom properties and not internal properties associated with the exchange in camel?

Reply:
I just realized that the whole problem was caused on the test side. The test is putting exchange into context to be accessible, I switched it to properties map and all works without any registration required. I'm sorry for wasting your time for this.

Reply:
I think if we expose all the properties (both custom and internal), it can not be accepted for security concerns.